examples/simple-chat/simple-chat.cpp

```diff
@@ -114,14 +114,15 @@ int main(int argc, char ** argv) {
             // check if we have enough space in the context to evaluate this batch
             int n_ctx = llama_n_ctx(ctx);
             int n_ctx_used = llama_memory_seq_pos_max(llama_get_memory(ctx), 0);
-            if (n_ctx_used + batch.n_tokens > n_ctx) {
+            if (n_ctx_used + batch.n_tokens >= n_ctx) {
```

Comment:
To be more precise, I think it would be better to add 1 to the value returned by `llama_memory_seq_pos_max`.
```C++
int n_ctx_used = llama_memory_seq_pos_max(llama_get_memory(ctx), 0) + 1; 
```